### ISSUE TITLE
Support customisation of logged payload

### ISSUE THREAD

**Issue by hdpe** [markhobson/rest-template-logger#6]
As discussed, it would be good to be able to customise the formatting of the request/response body before logging.

My particular use case is the need to obfuscate passwords sent in the request before writing to the log.

As a very first stab at this, how about allowing the `LoggingInterceptor` to be constructed with a `LoggingFormatter`, something like:

```java
public interface LoggingFormatter {
	
	String formatRequest(HttpRequest request, byte[] body);
	
	String formatResponse(ClientHttpResponse response);
}
```

with a default implementation:

```java
public class DefaultLoggingFormatter implements LoggingFormatter {
	
	@Override
	public String formatRequest(HttpRequest request, byte[] body) {
		// tmpl method to generate log line from URL, method, body...
	}
	
	@Override
	public String formatResponse(ClientHttpResponse response) {
		// tmpl method to generate log line from response status, body...
	}
	
	protected String formatBody(byte[] body, Charset charset) {
		return new String(body, charset);
	}
	
	protected Charset getCharset(HttpMessage message)
	{
		// ...
	}
}
```

This would support resolution of e.g. #2 as well. Then perhaps #1 via introducing something like a `supports(MediaType)` method to the formatter?

What do you think? Happy to prepare a PR following your feedback.

**Comment by markhobson** [comment]
I like it a lot! It certainly provides a good place to address the other issues as you say. Thanks Ryan :+1:

P.S. A pedant would say `LogFormatter` as it's not actually doing the logging.

**Comment by markhobson** [comment]
Thanks for the PR!

**Comment by jmart1** [comment]
Do you know when this version will be published to Maven Central?

**Comment by markhobson** [comment]
@jmart1 I've just released 2.0.0, so it should be there shortly. Thanks for the reminder!

**Comment by jmart1** [comment]
Yup its there.  Thank you!!!
